libcaf_net/caf/net/web_socket/framing.hpp

```diff
@@ -43,6 +43,11 @@ class CAF_NET_EXPORT framing : public octet_stream::upper_layer,
   /// Stored as currently active opcode to mean "no opcode received yet".
   static constexpr size_t nil_code = 0xFF;
 
+  // -- static utility functions -----------------------------------------------
+
+  static expected<std::pair<uint16_t, const_byte_span>>
+  decode_closing_payload(const_byte_span payload);
```

Comment:
I couldn't find a use for the actual result of this function outside of tests. It seems like we only really care about error vs. no error, right? In this case, we might consider just returning an `error` here to check for default-constructed (i.e., no-error) vs actual error:

```
  /// Checks whether ...
  /// @returns A default-constructed `error` if the payload is valid ...
  static error validate_closing_payload(const_byte_span payload);
```

Reply:
Currently there is no need to report the status code to the user, so we can do it this way for clarity.